packages/@lwc/ssr-compiler/src/__tests__/transmogrify.spec.ts

```diff
@@ -0,0 +1,130 @@
+import { parseModule } from 'meriyah';
+import { generate } from 'astring';
+import { beforeAll } from 'vitest';
+import { transmogrify } from '../transmogrify';
+import type { Program as EsProgram } from 'estree';
+
+const COMPILED_CMP = `
+    async function* tmpl(props, attrs, slottedContent, Cmp, instance) {
+      yield "<p";
+      yield stylesheetScopeTokenClass;
+      yield ">Hello</p>";
+    }
+
+    function* somethingElse() {
+      yield "doit";
+    }
+    async function* something() {
+      yield "foobar";
+      yield* somethingElse();
+    }
+    class Basic extends LightningElement {
+      static renderMode = "light";
+      async connectedCallback() {
+        for await (const thing of something()) {
+          console.log(thing);
+        }
+      }
+    }
+    const __REFLECTED_PROPS__ = [];
+    async function* generateMarkup(tagName, props, attrs, slotted) {
+      attrs = attrs ?? ({});
+      const instance = new Basic({
+        tagName: tagName.toUpperCase()
+      });
+      instance[SYMBOL__SET_INTERNALS](props, __REFLECTED_PROPS__, attrs);
+      instance.isConnected = true;
+      if (instance.connectedCallback) {
+        mutationTracker.enable(instance);
+        instance.connectedCallback();
+        mutationTracker.disable(instance);
+      }
+      const tmplFn = tmpl ?? fallbackTmpl;
+      yield \`<\${tagName}\`;
+      yield tmplFn.stylesheetScopeTokenHostClass ?? '';
+      yield* renderAttrs(instance, attrs);
+      yield '>';
+      yield* tmplFn(props, attrs, slotted, Basic, instance);
+      yield \`</\${tagName}>\`;
+    }
+`;
+
+describe('transmogrify', () => {
+    let COMPILED_CMP_SYNC: string;
+    let COMPILED_CMP_ASYNC: string;
+
+    beforeAll(() => {
+        const COMPILED_CMP_AST = parseModule(COMPILED_CMP, {
+            module: true,
+            next: true,
+        }) as EsProgram;
+        const COMPILED_CMP_AST_SYNC = transmogrify(COMPILED_CMP_AST, 'sync');
+        const COMPILED_CMP_AST_ASYNC = transmogrify(COMPILED_CMP_AST, 'async');
+        COMPILED_CMP_SYNC = generate(COMPILED_CMP_AST_SYNC);
+        COMPILED_CMP_ASYNC = generate(COMPILED_CMP_AST_ASYNC);
+    });
+
+    describe('in sync mode', () => {
+        test('generateMarkup is transformed into sync mode', () => {
+            expect(COMPILED_CMP_SYNC).not.toContain('async function* generateMarkup');
+            expect(COMPILED_CMP_SYNC).not.toContain('async function generateMarkup');
+            expect(COMPILED_CMP_SYNC).toContain('function generateMarkup($$emit');
+            expect(COMPILED_CMP_SYNC).not.toContain('yield* renderAttrs');
+            expect(COMPILED_CMP_SYNC).toContain('renderAttrs($$emit');
+            expect(COMPILED_CMP_SYNC).not.toContain('yield ">"');
+            expect(COMPILED_CMP_SYNC).not.toContain("yield '>'");
+            expect(COMPILED_CMP_SYNC).toContain('$$emit(">")');
```

Comment:
I think it would be easier to parse what should/n't be there if the `.not` / not `.not` were each grouped together. (And maybe each batched into an array to DRY the assertion. Or maybe not.)

Reply:
Let me add some whitespace in here. It was grouped (poorly) into 1) the thing I want to see, alongside 2) the thing I don't want to see. But that is not super obvious without whitepace.